src/pages/commits/CommitsWrapper.tsx

```diff
@@ -28,91 +36,130 @@ export const CommitsWrapper: React.FC<Props> = ({
 
+  const widths = versions.map((commit) => getCommitWidth(commit));
+  if (versions) {
     return (
-      <ProjectHealthWrapper>
-        <FlexRowContainer numCommits={versions.length}>
-          {versions.map(({ version, rolledUpVersions }) =>
-            version ? (
-              <ActiveCommit
-                key={version.id}
-                version={version}
-                chartType={chartType}
-                total={versionToGroupedTaskStatsMap[version.id].total}
-                max={max}
-                groupedTaskStats={
-                  versionToGroupedTaskStatsMap[version.id].stats
-                }
+      <>
+        <ChartWrapper>
+          <FlexRowContainer>
+            {versions.map((commit, i) => (
+              <CommitWrapper key={getCommitKey(commit)} width={widths[i]}>
+                <RenderCommitsChart
+                  hasTaskFilter={hasTaskFilter}
+                  commit={commit}
+                  chartType={chartType}
+                  max={max}
+                  groupedResult={versionToGroupedTaskStatsMap}
+                />
+              </CommitWrapper>
+            ))}
+          </FlexRowContainer>
+          <Grid numDashedLine={5} />
+          <AbsoluteContainer>
+            <ChartToggle
+              currentChartType={chartType}
+              onChangeChartType={onChangeChartType}
+            />
+          </AbsoluteContainer>
+        </ChartWrapper>
+        <StickyContainer>
+          <FlexRowContainer>
+            {versions.map((commit, i) => (
+              <CommitWrapper key={getCommitKey(commit)} width={widths[i]}>
+                <RenderCommitsLabel commit={commit} hasFilters={hasFilters} />
+              </CommitWrapper>
+            ))}
+          </FlexRowContainer>
+        </StickyContainer>
+        <FlexRowContainer>
+          {versions.map((commit, i) => (
+            <CommitWrapper key={getCommitKey(commit)} width={widths[i]}>
+              <RenderCommitsBuildVariants
+                commit={commit}
                 hasTaskFilter={hasTaskFilter}
               />
-            ) : (
-              <ColumnContainer key={rolledUpVersions[0].id}>
-                <InactiveCommitLine />
-                <InactiveCommits
-                  hasFilters={hasFilters}
-                  rolledUpVersions={rolledUpVersions}
-                />
-              </ColumnContainer>
-            )
-          )}
+            </CommitWrapper>
+          ))}
         </FlexRowContainer>
-        <Grid numDashedLine={5} />
-        <ChartToggle currentChartType={chartType} />
-      </ProjectHealthWrapper>
+      </>
     );
   }
   return <NoResults data-cy="no-commits-found">No commits found</NoResults>;
 };
 
+const getCommitWidth = (commit: Commit) => {
+  const { version, rolledUpVersions } = commit;
+  if (version) {
+    return 200;
+  }
+  if (rolledUpVersions) {
+    return 64;
+  }
+  throw new Error("Commit type not found");
+};
```

Comment:
It seems like `ChartContainer` in `CommitChart.tsx` is inheriting this `200px` width and it's causing a bug where the`Tooltip` is pretty far away from the actual bars in the graph (differs from current Spruce behavior)


<img width="1506" alt="Screen Shot 2022-01-21 at 4 51 47 PM" src="https://user-images.githubusercontent.com/47064971/150605514-d4cbef1e-5409-419a-b4c6-fbfa3b0011cd.png">

I don't know if this is the best solution but adding `width: fit-content` to the `ChartContainer` does fix this issue!

<img width="1503" alt="Screen Shot 2022-01-21 at 5 00 32 PM" src="https://user-images.githubusercontent.com/47064971/150605928-6a04d8ff-6cb8-4fab-a9c3-a9079c84a7e9.png">